stdlib/filename.ml

```diff
@@ -361,3 +361,13 @@ let open_temp_file ?(mode = [Open_text]) ?(perms = 0o600)
     with Sys_error _ as e ->
       if counter >= 1000 then raise e else try_name (counter + 1)
   in try_name 0
+
+let temp_dir ?(temp_dir = Domain.DLS.get current_temp_dir_name) prefix suffix =
+  let rec try_name counter =
+    let name = temp_file_name temp_dir prefix suffix in
+    try
+      Sys.mkdir name 0o700;
+      name
+    with Sys_error _ as e ->
+      if counter >= 1000 then raise e else try_name (counter + 1)
+  in try_name 0
```

Comment:
The retry logic is going to trigger if `temp_dir` does not exist. I don't think this is a good idea, check it for existence first.

Reply:
This is the same logic as in `temp_file`, so if doing this change it should be done in both functions. (I am not opposed to it.)

Reply:
Well let's avoid the future issue: "`Filename.temp_{file,dir}` takes a long time to fail if `temp_dir` does not exist".

Reply:
I don't think we should be adding such a check. The temp directory not existing is only one possible way for the function to fail after iterating all the way, but running out of space or inodes is more likely and can't be checked in such a way.

A better behaved function would be better, but it's not a big deal to spend 2ms failing because the whole program is likely to crash as a result anyway.

Maybe the doc can be tweaked but I'm in favor of proceeding as is otherwise.